src/jack.c

```diff
@@ -491,3 +510,68 @@ jalv_backend_activate_port(Jalv* jalv, uint32_t port_index)
 	}
 #endif
 }
+
+int
+jack_initialize(jack_client_t* const client, const char* const load_init)
+{
+	Jalv* const jalv = (Jalv*)calloc(1, sizeof(Jalv));
+	if (!jalv) {
+		return -1;
+	}
+
+	if (!(jalv->backend = (JalvBackend*)calloc(1, sizeof(JalvBackend)))) {
+		free(jalv);
+		return -1;
+	}
+
+	const size_t args_len = strlen(load_init);
+	if (args_len > JACK_LOAD_INIT_LIMIT) {
+		fprintf(stderr, "error: Too many arguments given\n");
+		free(jalv);
+		return -1;
+	}
+
+	jalv->backend->client             = client;
+	jalv->backend->is_internal_client = true;
+
+	/* Build full command line with "program" name for building argv */
+	const size_t cmd_len = strlen("jalv ") + args_len;
+	char* const  cmd     = (char*)calloc(cmd_len + 1, 1);
+	strcat(cmd, "jalv ");
+	strcat(cmd, load_init);
+
+	/* Build argv */
+	int         argc = 0;
+	char**      argv = (char**)malloc(sizeof(char**));
+	char*       tok  = cmd;
+	for (size_t i = 0; i <= cmd_len; ++i) {
+		if (isspace(cmd[i]) || !cmd[i]) {
+			argv = (char**)realloc(argv, sizeof(char*) * ++argc);
+			cmd[i] = '\0';
+			argv[argc - 1] = tok;
+			tok = cmd + i + 1;
+		}
+	}
+
+	const int err = jalv_open(jalv, argc, argv);
+	if (err) {
+		free(jalv);
```

Comment:
free(jalv->backend) is missing here

Reply:
`jalv_open()` should always call `jalv_close()` in case of an error. Otherwise the jack client will also not be closed